test/e2e/ingress_basicauth.go

```diff
@@ -153,4 +153,236 @@ var _ = Describe("IngressWithBasicAuth", func() {
 		})
 	})
+
+	Describe("Service Auth", func() {
+		BeforeEach(func() {
+			meta, err := f.Ingress.CreateResourceWithServiceAuth(secret)
+			Expect(err).NotTo(HaveOccurred())
+
+			ing.Spec.Rules = []api.IngressRule{
+				{
+					IngressRuleValue: api.IngressRuleValue{
+						HTTP: &api.HTTPIngressRuleValue{
+							Paths: []api.HTTPIngressPath{
+								{
+									Path: "/testpath",
+									Backend: api.HTTPIngressBackend{
+										IngressBackend: api.IngressBackend{
+											ServiceName: meta.Name,
+											ServicePort: intstr.FromInt(80),
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			}
+		})
+
+		It("Should response HTTP", func() {
+			By("Getting HTTP endpoints")
+			eps, err := f.Ingress.GetHTTPEndpoints(ing)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(len(eps)).Should(BeNumerically(">=", 1))
+
+			err = f.Ingress.DoHTTPStatus(framework.NoRetry, ing, eps, "GET", "/testpath", func(r *testserverclient.Response) bool {
+				return Expect(r.Status).Should(Equal(http.StatusUnauthorized))
+			})
+			Expect(err).NotTo(HaveOccurred())
+
+			err = f.Ingress.DoHTTPWithHeader(
+				framework.NoRetry,
+				ing,
+				eps,
+				"GET",
+				"/testpath",
+				map[string]string{
+					"Authorization": "Basic wrongPass",
+				},
+				func(r *testserverclient.Response) bool {
+					return Expect(r.Status).Should(Equal(http.StatusOK)) &&
+						Expect(r.Method).Should(Equal("GET")) &&
+						Expect(r.Path).Should(Equal("/testpath"))
+				},
+			)
+			Expect(err).To(HaveOccurred())
+
+			err = f.Ingress.DoHTTPWithHeader(
+				framework.NoRetry,
+				ing,
+				eps,
+				"GET",
+				"/testpath",
+				map[string]string{
+					"Authorization": "Basic Zm9vOmJhcg==",
+				},
+				func(r *testserverclient.Response) bool {
+					return Expect(r.Status).Should(Equal(http.StatusOK)) &&
+						Expect(r.Method).Should(Equal("GET")) &&
+						Expect(r.Path).Should(Equal("/testpath"))
+				},
+			)
+			Expect(err).NotTo(HaveOccurred())
+
+			err = f.Ingress.DoHTTPWithHeader(
+				framework.NoRetry,
+				ing,
+				eps,
+				"GET",
+				"/testpath",
+				map[string]string{
+					"Authorization": "Basic YXV0aDItZm9vOmJhcg==",
+				},
+				func(r *testserverclient.Response) bool {
+					return Expect(r.Status).Should(Equal(http.StatusOK)) &&
+						Expect(r.Method).Should(Equal("GET")) &&
+						Expect(r.Path).Should(Equal("/testpath"))
+				},
+			)
+			Expect(err).NotTo(HaveOccurred())
+
+			err = f.Ingress.DoHTTPWithHeader(
+				framework.NoRetry,
+				ing,
+				eps,
+				"GET",
+				"/testpath",
+				map[string]string{
+					"Authorization": "Basic amFuZTpndWVzdA==",
+				},
+				func(r *testserverclient.Response) bool {
+					return Expect(r.Status).Should(Equal(http.StatusOK)) &&
+						Expect(r.Method).Should(Equal("GET")) &&
+						Expect(r.Path).Should(Equal("/testpath"))
+				},
+			)
+			Expect(err).NotTo(HaveOccurred())
+		})
+	})
+
+	Describe("Both Global and Service Auth", func() {
```

Comment:
We need a test that checks that using the auth from service returns `StatusUnauthorized`, when both are in use.